File: vertx-kotlin-book/code-go/internal/migrate/migrate.go
// Package migrate applies the idempotent schema on startup. For a real system
// prefer a dedicated migration tool (golang-migrate, goose); this mirrors the
// Kotlin DbMigrator which runs the same SQL.
package migrate

import (
	"context"
	_ "embed"

	"github.com/jackc/pgx/v5/pgxpool"
)

//go:embed schema.sql
var schemaSQL string

func Run(ctx context.Context, pool *pgxpool.Pool) error {
	_, err := pool.Exec(ctx, schemaSQL)
	return err
}

File: vertx-kotlin-book/code-rust/src/migrate.rs
//! Applies the idempotent schema on startup. For a real system prefer a
//! dedicated migration tool (refinery, sqlx-migrate); this mirrors the Kotlin
//! DbMigrator / Go migrate package, which run the same SQL.

use deadpool_postgres::Pool;

const SCHEMA: &str = include_str!("../migrations/schema.sql");

pub async fn run(pool: &Pool) -> anyhow::Result<()> {
    let client = pool.get().await?;
    client.batch_execute(SCHEMA).await?;
    Ok(())
}
